src/assets/20-tech-terms/20-beginner-tech-terms.md

```diff
@@ -0,0 +1,141 @@
+
+## 17. API
+An `API (Application Programming Interface)` is a set of guidelines that allows different applications to communicate, and it provides an interface for developers to interact with other services. So we can picture it as a translator that helps two people speaking different languages understand each other.
+Key Features:
+- `Scalability`: Easy to extend and adapt for interoperation with other systems and applications
+- `Efficiency`: Can be embedded in other applications with great ease, thus giving you the freedom to utilize present functionality again instead of writing code from scratch.
+- `Security`: Ensures secure ways of accessing resources and a healthy trade of data.
+
+## 18. HTTP methods
+Simply, `HTTP methods` are actions and provide a way to indicate the desired action to be performed on a given resource within web server. They are crucial parts of the `HTTP protocol` that are used in communication on the web. Common `HTTP methods` are:
+- `GET` method: The `GET` method is designed to request a specific resource.
+- `POST` method: The POST method is designed to submit data to be processed by a specified resource. It is akin to completing and sending a form.
+- `PUT` method: The `PUT` method is designed to replace an entire resource with a new content. It's pretty much similar to editing a document.
+- `DELETE` method: When the `DELETE` method is triggered, the server removes the resource identified by that request. This is like deleting a file.
+
+**Example scenario:**
+
+When you open any webpage, the browser sends a `GET` request to its server to get its page data. When you fill out a form it actually is sending your data over with a `POST` request.
+
+## 19. Linter
+The process of analyzing the source code and reporting common syntactic and semantic errors or potential bugs is called `linting` and is done by using a `linter`.
+Think of a project you are working on using JavaScript. Even though you are writing the code yourself, how do you guarantee your code to be without any errors or faults and also follow coding standards? That's where a linter like `ESLint` is important. It automatically checks the quality of your code such as checking typos and applying coding guidelines. For example, it can warn when a variable is declared but never used or normalize quotes to either single or double.
+
+
+## 20. YAGNI
+`YAGNI (You Aren't Gonna Need It)` - It is a practice in extreme programming and other agile practices that tells not to implement any functionality until it is needed.
+YAGNI tells developers to only concentrate on features that are demanded in the present task. This is to prevent solutions from being built that will never be used, and keep a codebase clean during its life. There might come  some situations that you get `YAGNI` comment from seniors. :grin:
```

Comment:
```suggestion
YAGNI tells developers to only concentrate on features that are demanded in the present task. This is to prevent solutions from being built that will never be used, and keep a codebase clean during its life. There might be some situations when you get a `YAGNI` comment from seniors. :grin:
```